src/wallet/wallet.cpp

```diff
@@ -2032,11 +2032,17 @@ SpendableInputs CWallet::FindSpendableInputs(
 
     for (const auto& ivk : orchardIvks) {
         std::vector<OrchardNoteMetadata> incomingNotes;
-        orchardWallet.GetFilteredNotes(incomingNotes, ivk, true, true, true);
+        orchardWallet.GetFilteredNotes(incomingNotes, ivk, true, true);
 
         for (const auto& noteMeta : incomingNotes) {
+            if (IsOrchardSpent(noteMeta.GetOutPoint())) {
+                continue;
+            }
+
             auto mit = mapWallet.find(noteMeta.GetOutPoint().hash);
             if (mit != mapWallet.end() && mit->second.GetDepthInMainChain() >= minDepth) {
+                // Check whether any of the potential spends exist in the main chain or
+                // the mempool.
```

Comment:
This comment should have been removed when we fixed the outpoint vs inpoint confusion.
```suggestion
```